### ISSUE TITLE
from probabilities to logits in "sparse_categorical_crossentropy(target, output, from_logits=False)"

### ISSUE THREAD

**Issue by ZhangXiao96** [keras-team/keras#11801]
It seems that simply using _tf.log(output)_ in the 4th line cannot transform the outputs of softmax to logits. Did I just miss something important?

    if not from_logits:
        _epsilon = _to_tensor(epsilon(), output.dtype.base_dtype)
        output = tf.clip_by_value(output, _epsilon, 1 - _epsilon)
        output = tf.log(output)

    output_shape = output.get_shape()
    targets = cast(flatten(target), 'int64')
    logits = tf.reshape(output, [-1, int(output_shape[-1])])
    res = tf.nn.sparse_softmax_cross_entropy_with_logits(
        labels=targets,
        logits=logits)
